wolfcrypt/src/asn.c

```diff
@@ -35337,14 +35338,24 @@ int DecodeAsymKey_Assign(const byte* input, word32* inOutIdx, word32 inSz,
         if (GetMyVersion(input, inOutIdx, &version, inSz) < 0)
             return ASN_PARSE_E;
         if (version != 0) {
-            WOLFSSL_MSG("Unrecognized version of ED25519 private key");
+            WOLFSSL_MSG("Unrecognized version of private key");
             return ASN_PARSE_E;
         }
 
         if (GetAlgoId(input, inOutIdx, &oid, oidKeyType, inSz) < 0)
             return ASN_PARSE_E;
-        if (oid != (word32)keyType)
-            return ASN_PARSE_E;
+
+        if (inOutKeyType != NULL) {
```

Comment:
I would like to return BAD_FUNC_ARG inOutKeyType is NULL.
Should always check the OID - if expected value or using it to determine key type outside.

Reply:
done